apps/backend/src/modules/contributors/contributor.controller.ts

```diff
@@ -167,22 +181,22 @@ class ContributorController extends BaseController {
 	 */
 	private async findAll(
 		options: APIHandlerOptions<{
-			query: { projectId?: string } & PaginationQueryParameters;
+			query: ContributorGetAllRequestDto;
 		}>,
 	): Promise<APIHandlerResponse> {
-		if (options.query.projectId) {
-			const projectId = Number(options.query.projectId);
+		const { contributorName, hasHidden, page, pageSize, projectId } =
+			options.query;
 
-			return {
-				payload: await this.contributorService.findAllByProjectId({
-					projectId,
-				}),
-				status: HTTPCode.OK,
-			};
-		}
+		const query = {
+			...(contributorName ? { contributorName } : {}),
+			...(hasHidden ? { hasHidden: Boolean(hasHidden) } : {}),
+			...(page ? { page: Number(page) } : {}),
+			...(pageSize ? { pageSize: Number(pageSize) } : {}),
+			...(projectId ? { projectId: Number(projectId) } : {}),
+		};
```

Comment:
This will work only for `contributorName`, `hasHidden` and `projectId` since I've defined them as `field?: type | undefined`, and won't work for `pageSize` and `page` cause for them I used `Partial<..>` so their type is `field?: type` which means, that either field present in the object and have type `type` or it is absent at all -- `undefined` cannot be assigned to `page`/`pageSize`

So I can rewrite it only this way
```ts
const query = {
	contributorName,
	hasHidden: hasHidden ? Boolean(hasHidden) : undefined,
	projectId: projectId ? Number(projectId) : undefined,
	...(page ? { page: Number(page) } : {}),
	...(pageSize ? { pageSize: Number(pageSize) } : {}),
};
```
but this way it's a bit inconsistant (there are both `projectId: projectId ? Number(projectId) : undefined,` and `...(page ? { page: Number(page) } : {}),` in this object construction) and I'm not sure what looks more readable: this way or that current one

Reply:
let's leave it as it is, we don't want to pass unneeded query parameters

Reply:
Oh, wait. Sorry for confusing. This code is responsible not for sending query (frontend), but for retrieving it (on backend).
Here is the code where we choose what to send (`apps/frontend/src/modules/contributors/contributors-api.ts`)
![image](https://github.com/user-attachments/assets/0e4164f6-5d5c-435f-9df5-d71b0a39997b)

So what about now. Should I change this (_backend_) code?

P.s. Noticed, that I forget to pass contributorName to queryToSend. I'll fix it